kernel/src/actions/mod.rs

```diff
@@ -149,22 +157,21 @@ impl Protocol {
 #[derive(Debug, Clone, PartialEq, Eq, Schema)]
 #[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
 #[cfg_attr(not(feature = "developer-visibility"), visibility::make(pub(crate)))]
-// TODO need to have a way to always write some fields but not always read them
 struct CommitInfo {
     /// The time this logical file was created, as milliseconds since the epoch.
-    /// TODO should this be a Timestamp?
-    pub(crate) timestamp: i64,
+    pub(crate) timestamp: Option<i64>,
```

Comment:
Should this be a `u64`? There is no chance of this being negative?

Reply:
we don't impl Schema on `u64` - the `LONG` type is `i64`

Reply:
Yeah, this is a long time issue in that java doesn't have unsigned types. Somewhere back in our issue list is something about figuring this out :p